lib/OpenStackExtract/MiqOpenStackVm/MiqOpenStackCommon.rb

```diff
@@ -63,22 +63,28 @@ def create_image_from_snapshot(snapshot_id, disk_format)
     }
     volume = volume_service.volumes.new(volume_options)
     volume.save
-    begin
-      while volume.status != 'available'
-        sleep(10)
-        volume = volume_service.volumes.get(volume.id)
-      end
-      response = volume_service.action(volume.id, 'os-volume_upload_image' => {
-                                         :image_name => "Temp Image from #{snapshot.name}",
-                                         :disk_format => disk_format})
-      image_id = response.body["os-volume_upload_image"]["image_id"]
-      while image_service.images.get(image_id).status.downcase != 'active'
-        sleep(10)
-      end
-      return image_service.images.get(image_id)
-    ensure
-      volume.destroy
+
+    while volume.status != 'available'
+      sleep(10)
+      volume = volume_service.volumes.get(volume.id)
+    end
+
+    response = volume_service.action(volume.id, 'os-volume_upload_image' => {
+                                       :image_name => "Temp Image from #{snapshot.name}",
+                                       :disk_format => disk_format})
+    image_id = response.body["os-volume_upload_image"]["image_id"]
+
+    while image_service.images.get(image_id).status.downcase != 'active'
+      sleep(10)
     end
+
+    $log.debug "#{log_prefix}: Deleting temp volume #{volume.name} in #{volume.status} status"
+    volume.destroy
```

Comment:
```volume.destroy``` was removed form ```ensure``` block for few reasons:
* it can (easily) fail too, since it relies on code few lines above and OSP APIs, and this failure is not handled
* if method failed before reaching volume.destroy line, most propably the volume even does not exist or attempt for its deletion would fail too
* currently, volume.destroy in ensure block doesn't fullfil its task to delete volume in all cases